glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt

```diff
@@ -228,15 +232,31 @@ internal class MetricsPingScheduler(
             // the right time?
             isAfterDueTime(now) -> {
                 Log.i(LOG_TAG, "The 'metrics' ping is scheduled for immediate collection, ${now.time}")
+                // **IMPORTANT**
+                //
                 // The reason why we're collecting the "metrics" ping in the `Dispatchers.API`
                 // context is that we want to make sure no other metric API adds data before
                 // the ping is collected. All the exposed metrics API dispatch calls to the
                 // engines through the `Dispatchers.API` context, so this ensures we are enqueued
                 // before any other recording API call.
-                @Suppress("EXPERIMENTAL_API_USAGE")
-                Dispatchers.API.launch {
-                    // This addresses (2).
-                    collectPingAndReschedule(now)
+                //
+                // * Do not change `Dispatchers.API.executeTask` to `Dispatchers.API.launch` as
```

Comment:
Should we call out explicitly what executeTask does, namely it executes it right away (on the thread pool)?